### PR TITLE
Bump libraries in cli

### PR DESCRIPTION
**Pull request by praveenrewar**
#### What this PR does / why we need it:
- Bump k8s.io/client-go to v0.26.1
- Bump sig.k8s.io/controller-runtime to v0.14.5
- Bump carvel-dev/kapp-controller to v0.46.0

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by rohitagg2020 — COMMENTED** [review]
LGTM

File: cli/pkg/kctrl/local/reconciler.go [hunk 1]
```diff
@@ -213,6 +216,11 @@ func (o *Reconciler) newReconcilers(
 	refTracker := reftracker.NewAppRefTracker()
 	updateStatusTracker := reftracker.NewAppUpdateStatus()
 
+	kubeConfig := kubeconfig.NewKubeconfig(coreClient, runLog)
+	compInfo := componentinfo.NewComponentInfo(coreClient, kubeConfig, "dev")
+
```
Comment (100mik) [inline cli/pkg/kctrl/local/reconciler.go:221]: nit: remove new line

**Review by 100mik — COMMENTED** [review]
Very nitty but I am otherwise fine with the changes

File: cli/pkg/kctrl/local/reconciler.go (hunk 1 above)
Comment (praveenrewar) [reply cli/pkg/kctrl/local/reconciler.go:221]: I kept it on purpose :)
If you take a look at the above declarations above, they are also grouped in a similar fashion.

Comment (100mik) [reply cli/pkg/kctrl/local/reconciler.go:221]: Ah gotchya!

**Review by 100mik — APPROVED** [review]
Good to go!

**Pull request merged**